packages/block-editor/src/components/global-styles/image-settings-panel.js

```diff
@@ -0,0 +1,54 @@
+/**
+ * WordPress dependencies
+ */
+import {
+	__experimentalToolsPanel as ToolsPanel,
+	__experimentalToolsPanelItem as ToolsPanelItem,
+	ToggleControl,
+} from '@wordpress/components';
+import { __, _x } from '@wordpress/i18n';
+
+export function useHasImageSettingsPanel( name, settings ) {
+	return name === 'core/image' && settings?.lightbox?.allowEditing;
+}
+
+export default function ImageSettingsPanel( {
+	onChange,
+	userSettings,
+	settings,
+	panelId,
+} ) {
+	const resetLightbox = () => {
+		onChange( undefined );
+	};
+
+	const onChangeLightbox = ( newSetting ) => {
+		onChange( {
+			enabled: newSetting,
+		} );
+	};
+
+	return (
+		<>
+			<ToolsPanel
+				label={ _x( 'Settings', 'Image settings' ) }
+				resetAll={ resetLightbox }
+				panelId={ panelId }
+			>
+				<ToolsPanelItem
+					hasValue={ () => !! userSettings?.lightbox }
```

Comment:
Wouldn't this be more accurate?

```suggestion
					hasValue={ () => settings?.lightbox?.enabled }
```

If so, you could get rid of the second [`useGlobalSetting()`](https://github.com/WordPress/gutenberg/pull/54071/files#diff-657b8414b6d828dcb36993221d14155661d212bd121a7866cb7aa3f6534d9989R95) in [screen-block.js](https://github.com/WordPress/gutenberg/pull/54071/files#diff-657b8414b6d828dcb36993221d14155661d212bd121a7866cb7aa3f6534d9989).

Reply:
This was my first thougth as well, but it wouldn't work as expected!

The `hasValue` prop of the `ToolsItemPanel` controls whether the "RESET" button is enabled or not. We only want to enable it if there is a value in the `userSettings`, meaning that the user has set a value themselves in the Global Styles.

E.g.: If the default value is `true`, but the user hasn't set anything in the Global Styles the checkbox should be checked, but it shouldn't be possible to "RESET" it:

<img width="303" alt="Screenshot 2023-09-12 at 15 34 20" src="https://github.com/WordPress/gutenberg/assets/5417266/e5e87b32-cf6f-4927-8970-d0a2a09faf07">

---- 

I agree that it's not too intuitive though so I ll add a comment explaining the above 🙂

Reply:
Added the code comment in 4f0709b2e81b40fe45bfec986bd9182226977f26

Reply:
As I was testing things here, I found a bug related to this. It would be fixed by comparing the value with the theme.json origin default value instead of checking the user origin, but it's pretty minor, so I'm happy to leave this as a follow-up so we can get this merged before the feature freeze.

1. Start with a clean `wp_global_styles` CPT either by deleting the one for the theme you're using in your database or using a new theme.
2. See that it isn't possible to reset the image lightbox setting.
3. Go to the layout settings and modify one of the values. Save the change.
4. Go back to the image lightbox settings and see that the reset option is now available.

https://github.com/WordPress/gutenberg/assets/5129775/b11db8ec-f204-49aa-8521-0b617cbbeec5

What happens when you try to change anything in the settings part of theme.json, the entire settings object is saved in the `wp_global_styles` CPT. In the code here, you're checking for the value from this saved object to be truthy, and because all defaults are written to the database, not just the ones changed, setting any of them will trigger all of them to have their default value in the user settings.

I see this as a side-effect of trying to store things that aren't Gutenberg UI settings (`layout.contentWidth`, `layout.wideWidth`, and `lightbox.enabled`) in the settings object. So I think it would be better to move these values out of the settings rather than updating the code to be more specific about which settings are saved.